IdentityCore/src/cache/accessor/MSIDDefaultTokenCacheAccessor.m

```diff
@@ -87,8 +90,13 @@ - (BOOL)saveTokensWithConfiguration:(MSIDConfiguration *)configuration
     result = [self saveIDTokenWithConfiguration:configuration response:response context:context error:error];
 
     if (!result) return result;
-
-    // Save SSO state (refresh token and account)
+    
+    //Save App metadata
+    result = [self saveAppMetadataWithConfiguration:configuration response:(MSIDAADTokenResponse *)response context:context error:error];
```

Comment:
why do you need to cast response to MSIDAADTokenResponse here? What if it's not MSIDAADTokenResponse?

Reply:
removed. This was before I moved to using the factory method.